Description: Remove -Wc++14-compat
Forwarded: not-needed
Origin: upstream, https://github.com/google/ruy/pull/351
Bug-Debian: https://bugs.debian.org/1114362

--- a/ruy/CMakeLists.txt
+++ b/ruy/CMakeLists.txt
@@ -3,9 +3,9 @@
 #   cmake/bazel_to_cmake.sh
 
 if(CMAKE_SYSTEM_NAME STREQUAL Windows)
-  set(ruy_0_Wall_Wcxx14_compat_Wextra_Wundef "")
+  set(ruy_0_Wall_Wextra_Wundef "")
 else()
-  set(ruy_0_Wall_Wcxx14_compat_Wextra_Wundef "-Wall;-Wextra;-Wc++14-compat;-Wundef")
+  set(ruy_0_Wall_Wextra_Wundef "-Wall;-Wextra;-Wundef")
 endif()
 
 if(CMAKE_SYSTEM_PROCESSOR STREQUAL arm)
@@ -26,7 +26,7 @@
   HDRS
     trace.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -43,7 +43,7 @@
   HDRS
     platform.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
 )
@@ -64,7 +64,7 @@
   HDRS
     check_macros.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
 )
@@ -75,7 +75,7 @@
   SRCS
     check_macros_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -89,7 +89,7 @@
   HDRS
     opt_set.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
 )
@@ -100,7 +100,7 @@
   HDRS
     time.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
 )
@@ -119,7 +119,7 @@
   HDRS
     wait.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   LINKOPTS
@@ -134,7 +134,7 @@
   SRCS
     wait_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   LINKOPTS
@@ -151,7 +151,7 @@
   HDRS
     size_util.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -164,7 +164,7 @@
   SRCS
     size_util_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -180,7 +180,7 @@
   HDRS
     tune.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -199,7 +199,7 @@
   HDRS
     system_aligned_alloc.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
 )
@@ -212,7 +212,7 @@
   HDRS
     prepacked_cache.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -227,7 +227,7 @@
   SRCS
     tune_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -242,7 +242,7 @@
   SRCS
     prepacked_cache_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -265,7 +265,7 @@
   HDRS
     allocator.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -280,7 +280,7 @@
   SRCS
     allocator_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -294,7 +294,7 @@
   HDRS
     side_pair.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -309,7 +309,7 @@
   HDRS
     block_map.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -328,7 +328,7 @@
   SRCS
     block_map_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -348,7 +348,7 @@
   HDRS
     blocking_counter.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   LINKOPTS
@@ -367,7 +367,7 @@
   HDRS
     thread_pool.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   LINKOPTS
@@ -388,7 +388,7 @@
   HDRS
     cpu_cache_params.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
 )
@@ -427,7 +427,7 @@
   HDRS
     cpuinfo.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_4_Wno_undef}
@@ -445,7 +445,7 @@
   HDRS
     path.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   PUBLIC
@@ -462,7 +462,7 @@
   HDRS
     denormal.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   PUBLIC
@@ -474,7 +474,7 @@
   HDRS
     performance_advisory.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   PUBLIC
@@ -486,7 +486,7 @@
   HDRS
     matrix.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   PUBLIC
@@ -500,7 +500,7 @@
   SRCS
     matrix_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -514,7 +514,7 @@
   HDRS
     mul_params.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   PUBLIC
@@ -529,7 +529,7 @@
   SRCS
     mul_params_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -543,7 +543,7 @@
   HDRS
     mat.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -558,7 +558,7 @@
   HDRS
     asm_helpers.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -573,7 +573,7 @@
   HDRS
     apply_multiplier.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -587,7 +587,7 @@
   SRCS
     apply_multiplier_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -602,7 +602,7 @@
   HDRS
     kernel_common.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -626,7 +626,7 @@
   HDRS
     pack_common.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -649,7 +649,7 @@
   HDRS
     kernel_arm.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -675,7 +675,7 @@
   HDRS
     pack_arm.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -706,7 +706,7 @@
   HDRS
     kernel_x86.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_7_mavx512bw_mavx512cd_mavx512dq_mavx512f_mavx512vl_arch_AVX512}
@@ -730,7 +730,7 @@
   HDRS
     pack_x86.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_7_mavx512bw_mavx512cd_mavx512dq_mavx512f_mavx512vl_arch_AVX512}
@@ -753,7 +753,7 @@
   HDRS
     have_built_path_for.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_7_mavx512bw_mavx512cd_mavx512dq_mavx512f_mavx512vl_arch_AVX512}
@@ -778,7 +778,7 @@
   HDRS
     kernel_x86.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_8_mavx2_mfma_arch_AVX2}
@@ -802,7 +802,7 @@
   HDRS
     pack_x86.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_8_mavx2_mfma_arch_AVX2}
@@ -825,7 +825,7 @@
   HDRS
     have_built_path_for.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_8_mavx2_mfma_arch_AVX2}
@@ -850,7 +850,7 @@
   HDRS
     kernel_x86.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_9_mavx_arch_AVX}
@@ -874,7 +874,7 @@
   HDRS
     pack_x86.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_9_mavx_arch_AVX}
@@ -897,7 +897,7 @@
   HDRS
     have_built_path_for.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     ${ruy_9_mavx_arch_AVX}
@@ -912,7 +912,7 @@
   HDRS
     kernel.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -942,7 +942,7 @@
   HDRS
     pack.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -982,7 +982,7 @@
   HDRS
     context.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   PUBLIC
@@ -1004,7 +1004,7 @@
   SRCS
     context_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1033,7 +1033,7 @@
     ctx.h
     ctx_impl.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1058,7 +1058,7 @@
   HDRS
     context_get_ctx.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1072,7 +1072,7 @@
   SRCS
     ctx_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1088,7 +1088,7 @@
   HDRS
     trmul_params.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1107,7 +1107,7 @@
   HDRS
     trmul.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1139,7 +1139,7 @@
   HDRS
     prepare_packed_matrices.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1158,7 +1158,7 @@
   HDRS
     create_trmul_params.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1183,7 +1183,7 @@
   HDRS
     validate.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1201,7 +1201,7 @@
   HDRS
     frontend.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1228,7 +1228,7 @@
     path.h
     ruy.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   PUBLIC
@@ -1252,7 +1252,7 @@
   SRCS
     perchannel_buffers_reallocation_test.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1274,7 +1274,7 @@
   HDRS
     pmu.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
@@ -1287,7 +1287,7 @@
   HDRS
     reference_mul.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   PUBLIC
@@ -1310,7 +1310,7 @@
   HDRS
     test.h
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   LINKOPTS
@@ -1340,7 +1340,7 @@
   SRCS
     benchmark.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=f32
@@ -1359,7 +1359,7 @@
   SRCS
     benchmark.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=u8
@@ -1378,7 +1378,7 @@
   SRCS
     benchmark.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1397,7 +1397,7 @@
   SRCS
     benchmark.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1416,7 +1416,7 @@
   SRCS
     benchmark.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=u8
@@ -1435,7 +1435,7 @@
   SRCS
     benchmark.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1454,7 +1454,7 @@
   SRCS
     benchmark.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1473,7 +1473,7 @@
   SRCS
     benchmark.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i16
@@ -1491,7 +1491,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=f32
@@ -1509,7 +1509,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=f64
@@ -1527,7 +1527,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=f32
@@ -1545,7 +1545,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=u8
@@ -1563,7 +1563,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1581,7 +1581,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1599,7 +1599,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=u8
@@ -1617,7 +1617,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1635,7 +1635,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1653,7 +1653,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1671,7 +1671,7 @@
   SRCS
     test_fast.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i16
@@ -1689,7 +1689,7 @@
   SRCS
     test_slow.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=f32
@@ -1709,7 +1709,7 @@
   SRCS
     test_slow.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=u8
@@ -1729,7 +1729,7 @@
   SRCS
     test_slow.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1749,7 +1749,7 @@
   SRCS
     test_slow.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=u8
@@ -1769,7 +1769,7 @@
   SRCS
     test_slow.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1789,7 +1789,7 @@
   SRCS
     test_slow.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i8
@@ -1809,7 +1809,7 @@
   SRCS
     test_slow.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
     -DRUY_TEST_LHSSCALAR=i16
@@ -1829,7 +1829,7 @@
   SRCS
     test_overflow_dst_zero_point.cc
   COPTS
-    ${ruy_0_Wall_Wcxx14_compat_Wextra_Wundef}
+    ${ruy_0_Wall_Wextra_Wundef}
     ${ruy_1_mfpu_neon}
     ${ruy_2_O3}
   DEPS
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,7 @@
 cmake_policy(SET CMP0048 NEW)
 project(ruy CXX)
 cmake_minimum_required(VERSION 3.13)  # Copied from IREE
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 
 include(GNUInstallDirs)
 
